<launch>

  <group ns="rocket">
    <rosparam file="$(find tvc_simulator)/config/prototypes/drone/rocket_parameters.yaml" />
  </group>

  <group ns="environment">
    <rosparam file="$(find tvc_simulator)/config/prototypes/drone/environment_parameters.yaml" />
  </group>

  <group ns="perturbation">
    <rosparam file="$(find tvc_simulator)/config/perturbations_parameters.yaml" />
  </group>

  <group ns="visualization">
    <rosparam file="$(find tvc_simulator)/config/prototypes/drone/visualization_parameters.yaml" />
  </group>

  <group ns="mpc">
    <rosparam file="$(find tvc_simulator)/config/prototypes/drone/mpc_parameters.yaml" />
  </group>
''
<!--  <node name="actuators_tests" pkg="tvc_simulator" type="actuators_tests.py" output="screen"/>-->

  <node name="control" pkg="tvc_simulator" type="drone_control" output ="screen"/>
  <node name="control_interface" pkg="tvc_simulator" type="control_interface" cwd = "node"/>
<!--  <node name="guidance" pkg="tvc_simulator" type="drone_guidance" output ="screen"  />-->
  <node name="fast_integrator" pkg="tvc_simulator" type="fast_integrator" cwd = "node" output ="screen"/>
  <node name="disturbance" pkg="tvc_simulator" type="disturbance.py" cwd = "node"/>
  <node name="data_fusion" pkg="tvc_simulator" type="data_fusion" cwd = "node"/>
  <node name="time_keeper" pkg="tvc_simulator" type="time_keeper" cwd = "node"/>
<!--  <node name="aerodynamic" pkg="tvc_simulator" type="aerodynamic.py" cwd = "node"  output ="screen"/>-->

  <node name="visualization" pkg="tvc_simulator" type="visualization" />
  <node name="rocket_state" pkg="rostopic" type="rostopic" args="echo /rocket_state"  />
  <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find tvc_simulator)/rviz/drone_GUI.perspective"/>

  <node pkg="rosbag" type="record" name="recorder" args="record -a -O $(find tvc_simulator)/log/log.bag"/>

</launch>


